sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -591,6 +591,15 @@ public boolean anyNull() {
     return BitSetMethods.anySet(baseObject, baseOffset, bitSetWidthInBytes / 8);
   }
 
+  public boolean allNull() {
```

Comment:
IMHO, lets keep the implementation very "Single key" focussed. So my understanding here is that this expression is only called for "Key Row". That "Key row" should have only a single column, Right ? ie, can `numFields > 1` ?

If not, perhaps we can reword this and/or inline it. It's only used in two/three places so perhaps okay to duplicate given that we are modifying a very core class `UnsafeRow` here.

Reply:
yes, my original idea of using allNull is for future extension of supporting multi-column.
maybe i should be update this in next PR and leave necessary comment for now to make it "Single key" focussed.
got it.

Reply:
you're right! let's make all these allNull change into next PR.